src/executables/javasp.cpp

```diff
@@ -282,6 +305,82 @@ main (int argc, char *argv[])
   return status;
 }
 
+static void javasp_signal_handler (int sig)
+{
+  JAVASP_SERVER_INFO jsp_info = JAVASP_SERVER_INFO_INITIALIZER;
+
+  if (os_set_signal_handler (sig, SIG_DFL) == SIG_ERR)
+    {
+      return;
+    }
+
+  if (is_signal_handling == true)
+    {
+      return;
+    }
+
+  int status = javasp_get_server_info (db_name, jsp_info); // if failed,
+  if (status == NO_ERROR && jsp_info.pid != JAVASP_PID_DISABLED)
+    {
+      (void) envvar_bindir_file (executable_path, PATH_MAX, UTIL_JAVASP_NAME);
+      if (command.compare ("running") != 0 || db_name.empty ())
+	{
+	  return;
+	}
+
+      if (running_info.pid == jsp_info.pid && running_info.port == jsp_info.port)
+	{
+	  is_signal_handling = true;
+	}
+      else
+	{
+	  return;
+	}
+
+      if (!is_signal_handling) // check again
+	{
+	  return;
+	}
+
+      int pid = getpid ();
+      std::string err_msg;
+      void *addresses[64];
+      int nn_addresses = backtrace (addresses, sizeof (addresses) / sizeof (void *));
+      char **symbols = backtrace_symbols (addresses, nn_addresses);
+
+      err_msg += "pid (";
+      err_msg += std::to_string (pid);
+      err_msg += ")\n";
+
+      for (int i = 0; i < nn_addresses; i++)
+	{
+	  err_msg += symbols[i];
+	  if (i < nn_addresses - 1)
+	    {
+	      err_msg += "\n";
+	    }
+	}
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_SP_SERVER_CRASHED, 1, err_msg.c_str ());
+
```

Comment:
It would be a good idea to free the symbols allocated memory by bracktrace_symbol(), but you don't have to.

Reply:
Thank you.
```suggestion
      free(symbols);
      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_SP_SERVER_CRASHED, 1, err_msg.c_str ());
```